### PR TITLE
Create Virtual Network PS Script 

### LINKED ISSUE #20: [Task 2]  Implement script to create Virtual Network
This task is to implement a script to create a Virtual Network
| Date | Activity| Time |
| :---:        |     :---:      |     :---: |
| 1/12   | Created task on Github    | 5 min    |
| 1/13    | Setup VSC, and research how to create scripts  | 2 hours    |
| 1/15    | Group Meeting, and created preliminary script  | 2 hours   |
| 1/16   |  Research through documentation and created working script | 2 hours   |
| 1/17   |  Reviewed class videos and finished script  | 2 hours   |
| 1/18  |  Group Meeting and added finishing changes to script, created pull request  | 3 hours   |

### PR DESCRIPTION
**Pull request by sunshinejen**
Task #20 
This script creates a Virtual Network with one subnet
1. Open Powershell
2. Make sure you navigate to where script is located in the *automation/vnet* directory and enter *`./deploy-vnet.ps1`*
3. Enter information that is prompted, you will need an existing Azure subscription and resource group name
4. You will be asked to supply values for the following parameters:
RGName: *Name of your existing resource group*
Location: *Example: westus2*
VNetName: *Name of your vnet*
VNetPrefix: *Example: 10.0.0.0/16*
SubNetName: *Name of your subnet*
SubVNetPrefix: *Example: 10.0.0.0/24*
5. On Azure confirm Virtual Network has been created with your given parameters
6. If you are testing make sure to delete any created resources

### CODE REVIEW

File: automation/vnet/deploy-vnet.ps1 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+.PARAMETER SubNetName
+    Required. Name of desired SubNet
+.PARAMETER SubVNetPrefix
+    Required. Desired address space for SubNet
+.NOTES
+  Version:        1.0
+  Author:         Jennifer Villacis
+  Creation Date:  01/15/21
+  Purpose: VNet creation Script
+#>
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1:24]: Very professional documentation

File: automation/vnet/deploy-vnet.ps1 [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    [Parameter(Mandatory=$true)]
+    [string]
+    $VNetPrefix,
+    [Parameter(Mandatory=$true)]
+    [string]
+    $SubNetName,
+    [Parameter(Mandatory=$true)]
+    [string]
+    $SubVNetPrefix
+   )
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1:46]: No hard coded parameters, well done

Comment (OyunominJuno) [reply automation/vnet/deploy-vnet.ps1:46]: Can use comments explaining the parameters name since it has abbreviations

**Comment by elb-dev** [comment]
Code works as it should. Should make sure to understand the documentation before running.

**Review by elb-dev — APPROVED** [review]
Works just fine.

**Comment by sunshinejen** [comment]
@elb-dev I updated the documentation! Thank you

**Review by OyunominJuno — APPROVED** [review]
This script is successfully creating a Virtual network on given subscription and resource group. No hard coded variables and can be used dynamically. Therefore good to merge

**Comment by OyunominJuno** [comment]
With two approvals, merging this pull request

**Pull request merged**